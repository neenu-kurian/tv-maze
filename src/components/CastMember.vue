<template>
  <div class="flex align-middle flex-col text-center gap-1 p-1">
    <div
      class="w-[130px] items-center h-[130px] rounded-full overflow-hidden shrink-0 border-2 border-white shadow-sm"
    >
      <AppImage :src="member.person.image?.original" :alt="member.person.name">
        <template #fallback>
          <div role="img" :aria-label="`No image available for ${member.person.name}`">
            <UserIcon />
          </div>
        </template>
      </AppImage>
    </div>
    <div>
      <div>{{ member.person.name }}</div>
      <div class="text-gray-500 text-[0.8rem]">{{ member.character.name }}</div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { UserIcon } from '@heroicons/vue/24/outline';
import AppImage from './AppImage.vue';
import type { CastMember } from '@/types/index.ts';

defineProps<{
  member: CastMember;
}>();
</script>
