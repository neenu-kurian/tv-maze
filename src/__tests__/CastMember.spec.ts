import { render } from '@testing-library/vue';
import CastMember from '@/components/CastMember.vue';
import { CastMemberProp } from './testdata.ts';
import { describe, it, expect } from 'vitest';

describe('HomeView.vue', () => {
  it('matches snapshot', () => {
    const { container } = render(CastMember, {
      props: {
        member: CastMemberProp,
      },
    });
    expect(container).toMatchSnapshot();
  });
});
