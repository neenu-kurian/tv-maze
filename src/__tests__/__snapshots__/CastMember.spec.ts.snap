// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex align-middle flex-col text-center gap-1 p-1"
  >
    <div
      class="w-[130px] items-center h-[130px] rounded-full overflow-hidden shrink-0 border-2 border-white shadow-sm"
    >
      <div
        class="h-full"
      >
        <img
          alt="Sonja Sohn"
          class="w-full h-full object-cover relative"
          src="https://static.tvmaze.com/uploads/images/original_untouched/4/10304.jpg"
        />
      </div>
    </div>
    <div>
      <div>
        Sonja Sohn
      </div>
      <div
        class="text-gray-500 text-[0.8rem]"
      >
        Det. Shakima "Kima" Greggs
      </div>
    </div>
  </div>
</div>
`;
